lib/lesmok/tags/cached_include.rb

```diff
@@ -12,58 +12,100 @@ class CachedInclude < ::Liquid::Include
-          if cache_val.blank?
-            if Lesmok.config.debugging?
-              Lesmok.logger.warn "[#{self.class}] No valid cache key given for '#{template_name}' template!"
-              Lesmok.logger.debug " -- No cache key given nor found for object: #{cached_on_obj.inspect.truncate(64)}"
-            end
-            if Lesmok.config.raise_errors?
-              raise ArgumentError.new("No valid cache key! given for '#{template_name}' template!")
-            end
-          end
+          cache_val = calculate_cache_key_for(context)
 
           return super unless cache_val.present?
 
-          ## Allow sub-scoping w/o manually creating cache key.
-          cache_subscope = context[@attributes['cache_scope']]
-          cache_val += ":SUBSCOPE-#{cache_subscope}" if cache_subscope.present?
+          template_name = context[@template_name]
 
-          expire_in = calculate_expiry(cached_on_obj, context[@attributes['expire_in']])
+          expire_in = calculate_expiry(context[@attributes['cache_on']], context[@attributes['expire_in']])
           cache_key = self.class.full_cache_key_for(cache_val, template_name)
           cache_store = select_cache_store_for(context)
-          Lesmok.logger.debug "[#{self.class}] Lookup #{cache_key} in #{cache_store}..." if Lesmok.config.debugging?
-          result = cache_store.fetch(cache_key, expires_in: expire_in) do
-            Lesmok.logger.debug "[#{self.class}] --- cache miss on #{cache_key} in #{cache_store}!" if Lesmok.config.debugging?
+
+          num_previous_errs = context.errors.size
+          result = perform_cached_inclusion_rendering_for(context, cache_store, cache_key, expire_in) do
             super
           end
 
-          if context.errors.present?
-            ::Lesmok.logger.debug " -- Liquid errors (#{context.errors.size}) seen in: #{@template_name}"
+          if context.errors.size > num_previous_errs
+            lesmok_logger.debug "[lesmok] -- Liquid errors (#{context.errors.size}) seen in: #{template_name}"
+            return serve_any_stale_cached_content(context, cache_key)
           end
 
           result
         rescue Exception => err
           log_exception(err, context)
-          ""
+          serve_any_stale_cached_content(context, cache_key) || ""
         end
 
-        def select_cache_store_for(context)
-          cache_store_name = context[@attributes['cache_store']]
+        def perform_cached_inclusion_rendering_for(context, cache_store, cache_key, expire_in)
+          lesmok_logger.debug "[#{self.class}] Lookup #{cache_key} in #{cache_store}..." if Lesmok.config.debugging?
+          result = cache_store.fetch(cache_key, expires_in: expire_in) do
+            lesmok_logger.debug "[#{self.class}] --- cache miss on #{cache_key} in #{cache_store}!" if Lesmok.config.debugging?
+            rendered_str = yield
+            if Lesmok.config.serve_stale_content? && rendered_str.present? && context.errors.empty?
```

Comment:
`context.errors.empty?` won't suffice if cached includes further down the chain has swallowed any errors...